src/components/Card/components/Header/Header.tsx

```diff
@@ -13,86 +13,85 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from '@emotion/styled';
+import React, { FC, ReactNode, MouseEvent, KeyboardEvent } from 'react';
 import { css } from '@emotion/core';
+import { Dispatch as TrackingProps } from '@sumup/collector';
 
-import CloseButton from '../../../CloseButton';
-import { childrenPropType } from '../../../../util/shared-prop-types';
+import styled, { StyleProps } from '../../../../styles/styled';
+import {
+  CloseButton,
+  CloseButtonProps,
+} from '../../../CloseButton/CloseButton';
 
-const baseStyles = ({ theme }) => css`
+export interface CardHeaderProps {
+  /**
+   * Heading to be shown.
+   */
+  children: ReactNode;
+  /**
+   * Text label for the close button for screen readers.
+   * Important for accessibility.
+   */
+  labelCloseButton?: string;
+  /**
+   * Additional data that is dispatched with the tracking event.
+   */
+  tracking?: TrackingProps;
+  /**
+   * Callback for the close button. If not specified, the button won't
+   * be shown.
+   */
+  onClose?: (event: MouseEvent | KeyboardEvent) => void;
+}
+
+const baseStyles = ({ theme }: StyleProps) => css`
   label: card__header;
   align-items: center;
   display: flex;
   justify-content: space-between;
   margin-bottom: ${theme.spacings.giga};
 `;
 
-const noHeadingStyles = ({ children }) =>
-  !children[0] &&
-  css`
-    label: card__header--no-heading;
-    justify-content: flex-end;
-  `;
```

Comment:
I assume you removed this because the `children` prop is required now? 

We still cannot assume that `children` will be defined since this isn't enforceable in JavaScript projects. 

Furthermore, should the `children` prop always be required? I can imagine a use case where the close button is needed, but no title for the card. What do you think?

Reply:
I fix the children, now It's not required 😄  and the style for this is back!